submitr/scripts/submit_metadata_bundle.py

```diff
@@ -52,26 +61,31 @@ def main(simulated_args_for_testing=None):
                         help=f"the submission protocol (default {DEFAULT_SUBMISSION_PROTOCOL!r})")
     parser.add_argument('--verbose', action="store_true", help="Debug output.", default=False)
     parser.add_argument('--debug', action="store_true", help="Debug output.", default=False)
-    parser.add_argument('--check', action="store_true",
-                        help="Sanity check file locally before submission.", default=False)
-    parser.add_argument('--check-only', action="store_true",
-                        help="Sanity check file locally ONLY (no submission).", default=False)
-    parser.add_argument('--nocheck', action="store_true",
-                        help="Do not sanity check file locally before submission.", default=False)
-    parser.add_argument('--validate-local', action="store_true", help="Synonym for --check.")
-    parser.add_argument('--validate-local-only', action="store_true", help="Synonym for --check-only.")
     args = parser.parse_args(args=simulated_args_for_testing)
 
     if args.directory:
         args.upload_folder = args.directory
     if args.subdirectories:
         args.subfolders = True
-    if args.check:
-        args.validate_local = True
-    if args.check_only:
-        args.validate_local_only = True
-    if args.validate_local_only:
+    if not (args.validate_only and (args.validate_local or args.validate_local_only)):
+        if args.validate_only:
+            args.validate_local = False
+        elif args.validate_local or args.validate_local_only:
+            args.validate_first = False
+            args.validate_only = False
+    if args.validate_only:
+        args.validate_first = False
+    if args.validate and not (args.validate_only or args.validate_local_only or
+                              args.validate_local_only or args.validate_first):
+        # If --validate is specified and no other validate related options are
+        # specified, then default to server-side and client-side validation.
         args.validate_local = True
+        args.validate_first = True
```

Comment:
When you get into complicated arg behavior like this I sometimes like to write a quick unit test + a helper function to deal with it, we don't want a scenario where this goes wrong (it looks right but still worth testing)

Reply:
Yes, sorry, should have keep up better with test, will do.